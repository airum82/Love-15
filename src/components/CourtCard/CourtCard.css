.court-card {
  background-color: rgba(229, 149, 0, .7);
}