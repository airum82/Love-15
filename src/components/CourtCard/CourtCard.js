import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import './CourtCard.css';

export const CourtCard = ({name, location, id, addFavoriteCourt, account }) => {
  const court = {
    name,
    location
  }
  return (
    <div className="court-card">
      <h2>{name}</h2>
      <NavLink to={`/court/${id}`}>
        <p>{location}</p>
      </NavLink>
      <button onClick={() => addFavoriteCourt(account.id, court)}>
        add to favorites
      </button>
    </div>
  )
}