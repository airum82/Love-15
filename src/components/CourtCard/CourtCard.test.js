import React from 'react';
import { CourtCard } from './CourtCard';
import { shallow } from 'enzyme';

describe('CourtCard', () => {

  it('should match snapShot upon render', () => {
    const wrapper = shallow(<CourtCard />);

    expect(wrapper).toMatchSnapshot();
  })
})