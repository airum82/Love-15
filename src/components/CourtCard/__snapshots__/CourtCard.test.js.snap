// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourtCard should match snapShot upon render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2 />,
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="/court/undefined"
>
          <p />
</NavLink>,
      ],
      "className": "court-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": <p />,
          "to": "/court/undefined",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2 />,
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/court/undefined"
>
            <p />
</NavLink>,
        ],
        "className": "court-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": <p />,
            "to": "/court/undefined",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
